fars/booking/views.py

```diff
@@ -209,6 +210,49 @@ def _is_unbookable(self, user, booking):
         return True, ''
 
 
+class TabletView(View):
+    template = 'tablet.html'
+    context = {}
+
+    def dispatch(self, request, bookable):
+        bookable_obj = get_object_or_404(Bookable, id_str=bookable)
+        if not bookable_obj.public and not request.user.is_authenticated:
+            return redirect('{}?next={}'.format(reverse('login'), request.path_info))
+        self.context['bookable'] = bookable_obj
+        self.context['errors'] = 0
+        return super().dispatch(request, bookable)
+
+    def get(self, request, bookable):
+        now = pytz.timezone(TIME_ZONE).localize(datetime.now())
+        booking = Booking()
+        booking.bookable = self.context['bookable']
+        self.context['bookform'] = BookingForm(instance=booking)
+        return render(request, self.template, context=self.context)
+
+
+    def post(self, request, bookable):
+        username = request.POST.get('username')
+        pw = request.POST.get('password')
+        user = authenticate(username=username, password=pw)
+        postdata = request.POST.copy()
+        try:
+            postdata['user'] = user.id
+        except AttributeError:
+            postdata['user'] = -1
+            self.context['errors'] = 1
+            self.context['credential_error'] = 1
```

Comment:
What the hell is going on here? what do these values mean? Are they mimicking booleans?

This was the unfinished part though? I'll re-review once it's ready then.

Reply:
This part will be clarified and done ready in a future PR. For now the functionality of this won't be used

Reply:
Please add a TODO or FIXME comment